github/repos_collaborators.go

```diff
@@ -94,7 +94,7 @@ type RepositoryAddCollaboratorOptions struct {
 	Permission string `json:"permission,omitempty"`
 }
 
-// AddCollaborator adds the specified GitHub user as collaborator to the given repo.
+// AddCollaborator invites the specified GitHub user as a collaborator to the given repo.
 //
 // GitHub API docs: https://developer.github.com/v3/repos/collaborators/#add-user-as-a-collaborator
 func (s *RepositoriesService) AddCollaborator(ctx context.Context, owner, repo, user string, opt *RepositoryAddCollaboratorOptions) (*Response, error) {
```

Comment:
Another thought I had: rename this to `InviteCollaborator` and revert `AddCollaborator` to the 'add directly' behavior. But this is yet another behavior change in `go-github` which might be confusing to clients. I'm open to feedback though.